components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/IdentityKeyStoreResolver.java

```diff
@@ -0,0 +1,599 @@
+ * Retrieve keystores, private keys, public keys and public certificates.
+ */
+public class IdentityKeyStoreResolver {
+
+    private static IdentityKeyStoreResolver instance = null;
+
+    private static ConcurrentHashMap<InboundProtocol, IdentityKeyStoreMapping>
+            keyStoreMappings = new ConcurrentHashMap<>();
+
+    // Hashmaps to store retrieved private keys and certificates.
+    // This will reduce the time required to read configs and load data from keystores everytime.
+    private static Map<String, Key> privateKeys = new ConcurrentHashMap<>();
+    private static Map<String, Certificate> publicCerts = new ConcurrentHashMap<>();
+
+    private static final Log LOG = LogFactory.getLog(IdentityKeyStoreResolver.class);
+
+    private IdentityKeyStoreResolver() {
+
+        parseIdentityKeyStoreMappingConfigs();
+    }
+
+    public static IdentityKeyStoreResolver getInstance() {
+
+        if (instance == null) {
+            instance = new IdentityKeyStoreResolver();
+        }
+        return instance;
+    }
+
+    /**
+     * Return Primary or tenant keystore according to given tenant domain.
+     *
+     * @param tenantDomain  Tenant domain
+     * @return Primary or tenant keystore
+     * @throws IdentityKeyStoreResolverException the exception in the IdentityKeyStoreResolver class
+     */
+    private KeyStore getKeyStore(String tenantDomain) throws IdentityKeyStoreResolverException {
+
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        KeyStoreManager keyStoreManager = KeyStoreManager.getInstance(tenantId);
+        try {
+            if (MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                // Get primary keystore from keyStoreManager
+                return keyStoreManager.getPrimaryKeyStore();
+            }
+
+            // Get tenant keystore from keyStoreManager
+            String tenantKeyStoreName = IdentityKeyStoreResolverUtil.buildTenantKeyStoreName(tenantDomain);
+            return keyStoreManager.getKeyStore(tenantKeyStoreName);
+        } catch (Exception e) {
+            throw new IdentityKeyStoreResolverException(
+                    ErrorMessages.ERROR_CODE_ERROR_RETRIEVING_TENANT_KEYSTORE.getCode(),
+                    String.format(ErrorMessages.ERROR_CODE_ERROR_RETRIEVING_TENANT_KEYSTORE.getDescription(),
+                            tenantDomain), e);
+        }
+    }
+
+    /**
+     * Return Primary, tenant or custom keystore.
+     *
+     * @param tenantDomain      Tenant domain
+     * @param inboundProtocol   Inbound authentication protocol of the application
+     * @return Primary, tenant or custom keystore
+     * @throws IdentityKeyStoreResolverException the exception in the IdentityKeyStoreResolver class
+     */
+    public KeyStore getKeyStore(String tenantDomain, InboundProtocol inboundProtocol)
+            throws IdentityKeyStoreResolverException {
+
+        if (tenantDomain == null || tenantDomain.isEmpty()) {
```

Comment:
```suggestion
        if (StringUtils.isEmpty(tenantDomain)) {
```
Let's use `StringUtils.isEmpty` where applicable.

Reply:
Fixed in https://github.com/wso2/carbon-identity-framework/pull/5794/commits/4f1ed236169ced422e2de09e5868eb593993c745